File: week07/mapreduce/hadoop/wordcount_java/src/main/java/wordcount/WordCountReducer.java
package wordcount;

import org.apache.hadoop.io.IntWritable;
import org.apache.hadoop.io.Text;
import org.apache.hadoop.mapreduce.Reducer;

import java.io.IOException;

public class WordCountReducer extends Reducer<Text, IntWritable, Text, IntWritable> {

  private IntWritable count = new IntWritable();

  @Override
  protected void reduce(Text key, Iterable<IntWritable> values, Context context) throws IOException, InterruptedException {
    int sum = 0;

    for (IntWritable v: values)
      sum += v.get();

    count.set(sum);

    context.write(key, count);
  } // end method reduce.

} // end class WordCountReducer.

File: week07/mapreduce/hadoop/wordcount_scala/src/main/scala/wordcount/WordCountReducer.scala
package wordcount

import java.lang

import org.apache.hadoop.io.{IntWritable, Text}
import org.apache.hadoop.mapreduce.Reducer

import collection.JavaConverters._

class WordCountReducer extends Reducer[Text, IntWritable, Text, IntWritable] {
  val count = new IntWritable()

  override def reduce(key: Text, values: lang.Iterable[IntWritable], context: Reducer[Text, IntWritable, Text, IntWritable]#Context): Unit = {
    count.set(values.asScala.map(x => x.get()).sum)
    context.write(key, count)
  } // end method reduce.

} // end class WordCountReducer
